test/src/test_encode_context.cpp

```diff
@@ -59,13 +59,6 @@ BOOST_AUTO_TEST_CASE(json_encode_context_should_return_same_address_for_multiple
   BOOST_CHECK(address_0 == address_2);
 }
 
-BOOST_AUTO_TEST_CASE(json_encode_context_should_advance_pointer_after_reservation) {
-  encode_context ctx(0);
-  ctx.advance(0); BOOST_CHECK(ctx.reserve(1024) == &ctx.data()[0]);
-  ctx.advance(1); BOOST_CHECK(ctx.reserve(1024) == &ctx.data()[1]);
-  ctx.advance(2); BOOST_CHECK(ctx.reserve(1024) == &ctx.data()[3]);
-}
-
```

Comment:
It should be possible to rewrite this test to be memory-safe by calling `data()` after the `reserve()` call. (I guess it could be called either before or after in the current code, depending on the compiler, since I don't think there's an ordering guarantee when evaluating the operands to `==`.)

Reply:
Done